.github/workflows/run-tests.yml

```diff
@@ -20,10 +20,24 @@ on:
 
 jobs:
   run-tests:
-    name: Run tests job
+    name: Run tests job matrix
 
     runs-on: ubuntu-latest
 
+    strategy:
+      matrix:
+        client-type: [go, http]
+        database-type: [badger-file, badger-memory]
+        mutation-type: [gql, collection-named, collection-save]
+        tests: [names, lens, cli]
+
```

Comment:
thought: I think we should also include the codecov stuff in this file, and delete the codecov workflow. I don't want to mark this as a todo though, as I'm still playing around with it and it can be done after easily enough.